### PR TITLE
*formulae: Avoid using `rm_rf`, `rm_f` or `rmtree`

### PR DESCRIPTION
**Pull request by issyl0**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

- Part of https://github.com/Homebrew/brew/pull/17705.

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]
Looks right!

**Review by woodruffw — APPROVED** [review]

File: Formula/c/ca-certificates.rb [hunk 1]
```diff
@@ -121,7 +121,7 @@ def macos_post_install
   end
 
   def linux_post_install
-    rm_f pkgetc/"cert.pem"
+    rm(pkgetc/"cert.pem")
```
Comment (carlocab) [inline Formula/c/ca-certificates.rb:124]: We should check that the file exists here, because it won't the first time this formula is installed.

**Review by carlocab — COMMENTED** [review]
We should probably be a bit more careful about the ones in `post_install` blocks, as those will be disruptive to more users.

File: Formula/c/ca-certificates.rb (hunk 1 above)
Comment (issyl0) [reply Formula/c/ca-certificates.rb:124]: Glad you're here! 😅

Comment (issyl0) [reply Formula/c/ca-certificates.rb]: How much "more careful" should we be? For all of the ones in the `post_install` blocks should I do a "if this file exists?" check?

Comment (MikeMcQuaid) [reply Formula/c/ca-certificates.rb]: @issyl0 I think anyone where we don't know specifically that it should be here: it's fine as-is.

**Comment by fxcoudert** [comment]
Two comments:
- `rm_f` is often used to mean “delete if it exists”, which may cover a difference in linux/macOS build process.
- I don't think this is just style issues, some of these may be breaking changes in formulas. We could bunch them in small batches, but they should be tested imho.

**Comment by Bo98** [comment]
For formulae, it might be an idea to just change `rm_f` to mean `rm if File.exist?` rather than the current ignore-all-errors.

**Comment by MikeMcQuaid** [comment]
> * which may cover a difference in linux/macOS build process.

I think these differences should be covered with `if`/`else` instead.

> For formulae, it might be an idea to just change `rm_f` to mean `rm if File.exist?`.

This would still be a breaking change. I think it'd be nicer to introduce a new method/API if we're going to do that rather than shadowing the `FileUtils` one.

**Comment by issyl0** [comment]
I re-generated this after the latest changes to the RuboCop.

**Comment by issyl0** [comment]
> I don't think this is just style issues, some of these may be breaking changes in formulas. We could bunch them in small batches, but they should be tested imho.

Hmmm. I'll bunch these in smaller batches, and test installing and testing each one locally. :-)

**Pull request closed without merging**